src/toolkits/activity_classification/class_registrations.cpp

```diff
@@ -17,6 +17,25 @@ namespace activity_classification {
 BEGIN_FUNCTION_REGISTRATION
 REGISTER_FUNCTION(_activity_classifier_prepare_data , "data" , "features" , "session_id" , "prediction_window" , "predictions_in_chunk" , "target")
 REGISTER_FUNCTION(_activity_classifier_prepare_data_verbose , "data" , "features" , "session_id" , "prediction_window" , "predictions_in_chunk" , "target")
+REGISTER_FUNCTION(activity_classifier::random_split_by_session, "data",
```

Comment:
I think `tc.extensions.random_split_by_session` might be a bit too generic. Could we try `tc.extensions.activity_classifier.random_split_by_session`? I believe you should be able to accomplish that using a macro called something like REGISTER_NAMED_FUNCTION, and the name "activity_classifier.random_split_by_session".

Reply:
I changed it to a REGISTER_NAMED_FUNCTION. With this change it compiles but does not work. I looked up the definition of the macro and I think it is suppose to work this way. Can you look at the change and let me know if there is something I am missing.